cmake/template/DLAFConfig.cmake.in

```diff
@@ -20,7 +20,6 @@ list(PREPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR})
 # ===== VARIABLES
 set(DLAF_WITH_OPENMP @DLAF_WITH_OPENMP@)
 set(DLAF_WITH_MKL @DLAF_WITH_MKL@)
-set(DLAF_WITH_MKL_LEGACY @DLAF_WITH_MKL_LEGACY@)
```

Comment:
shouldn't `DLAF_WITH_MKL` changed to `DLAF_WITH_MKL_INTERNAL`?

Reply:
Or not be set at all? Given that it has `INTERNAL` in the name I don't think we should expect users to rely on `DLAF_WITH_MKL_INTERNAL` to decide anything. We don't need it in `DLAFConfig.cmake` anymore.

If anything stays perhaps `set(DLAF_WITH_MKL @DLAF_WITH_MKL_INTERNAL@)` as that's consistent with the preprocessor definition https://github.com/eth-cscs/DLA-Future/blob/1758a6694729dc0a5c12e3b0b94eba4df54c1d04/src/CMakeLists.txt#L106.

What do you think?

Reply:
Good point! it shouldn't be needed anymore.

Reply:
https://github.com/eth-cscs/DLA-Future/pull/1149/commits/5f5bb92d996ec30505f07de56936c21d2330e35b